docs: update docs to reflect min requirements and better explain roles (#571)
CrowdStrike/ansible_collection_falcon: roles/falcon_configure/README.md

# crowdstrike.falcon.falcon_configure

> Configures the CrowdStrike Falcon Sensor. This role is focused mainly on configuring the Falcon Sensor on Linux and macOS. Windows is supported, but not as much functionality is currently available. The main difference is because a lot of the configuration options can be set during the installation of the sensor on Windows.

Reviewer: ```suggestion
Configures the CrowdStrike Falcon Sensor. This role is focused mainly on configuring the Falcon Sensor on Linux and macOS. Windows is supported, but not as much functionality is currently available  because of all the configuration options that can be set during the installation of the Window's sensor.
```

- When using API credentials `falcon_client_id` and `falcon_client_secret`
  - **Sensor Download** [read]

## Falcon Sensor Service

For Linux/macOS, the Falcon Sensor service is automatically started after the sensor is installed (*although it will be in a failed state*).

> [!WARNING]
> The service will be in a failed state until the CID is set and the sensor is restarted.

This role will restart the Falcon Sensor service after the CID is set, along with any other configuration options.

> 

Reviewer 2: This is almost more confusing than saying nothing. What is the point you're trying to get across?

Author: The point I'm trying to get across is that folks are confused by how the falcon-sensor is started.. the thing is, the sensor is always started by default, it's just in a failed state until you provide the CID and restart the sensor. It seems like people try to manage the falcon-sensor service manually since it's apparently not clear in here.

Any thoughts on properly wording that?

## Dependencies

- Privilege escalation (sudo) is required for this role to function properly.
- The Falcon Sensor must be installed

> [!IMPORTANT]

> > The role automatically handles privilege escalation so you do not need to set `become: true` in your playbook.

Reviewer 2: This appears to directly conflict with the first bullet above. I think you're saying, "you need to be able to escalate, but do not escalate the entire role."

Author: This statement translates to:
DO NOT SET BECOME for these roles!

Reviewer: How about something like:

```
- The Falcon Sensor must be installed
- Privilege escalation (sudo) is required for this role to function properly. Please make sure that it is configured in your environment. When the role is run, it will automatically handle privilege escalation so avoid setting `become` in your playbooks to prevent issues when the playbook is run.
```
